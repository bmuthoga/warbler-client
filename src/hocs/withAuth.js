import React, { Component } from 'react'
import { connect } from 'react-redux'

export default function withAuth(ComponentToBeRendered) {
  class Authenticate extends Component {
    constructor(props) {
      super(props)
      if (!props.isAuthenticated) {
        props.history.push('/signin')
      } 
    }

    UNSAFE_componentWillUpdate() {
      if (!this.props.isAuthenticated) {
        this.props.history.push('/signin')
      }
    }

    render() {
      return <ComponentToBeRendered { ...this.props } />
    }
  }

  const mapStateToProps = state => {
    return {
      isAuthenticated: state.currentUser.isAuthenticated
    }
  }

  return connect(mapStateToProps)(Authenticate)
}
