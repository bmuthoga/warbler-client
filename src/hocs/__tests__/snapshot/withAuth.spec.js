import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import withAuth from '../../withAuth'

describe('withAuth Snapshot Tests', () => {
  let WithAuthComponent
  let mockStore

  beforeEach(() => {
    mockStore = configureMockStore()
    const mockedComponent = jest.fn()
    WithAuthComponent = withAuth(mockedComponent)
  })

  it('should render the component with props.isAuthenticated equal to true', () => {
    const store = mockStore({
      currentUser: {
        isAuthenticated: true,
        user: {
          id: '7585jjjjgig85jtjv',
          username: 'testuser',
          profileImageUrl: 'https://lorempixel.com/200/200/'
        }
      }
    })
    const props = {
      history: {
        push: jest.fn()
      }
    }
    const wrapper = shallow(
      <WithAuthComponent { ...props } store={store} />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render the component with props.isAuthenticated equal to false', () => {
    const store = mockStore({
      currentUser: {
        isAuthenticated: false,
        user: {}
      }
    })
    const props = {
      history: {
        push: jest.fn()
      }
    }
    const wrapper = shallow(
      <WithAuthComponent { ...props } store={store} />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
