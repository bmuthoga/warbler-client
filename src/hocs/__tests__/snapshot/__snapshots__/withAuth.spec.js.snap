// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuth Snapshot Tests should render the component with props.isAuthenticated equal to false 1`] = `
<ContextProvider
  value={null}
>
  <Authenticate
    dispatch={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    isAuthenticated={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`withAuth Snapshot Tests should render the component with props.isAuthenticated equal to true 1`] = `
<ContextProvider
  value={null}
>
  <Authenticate
    dispatch={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    isAuthenticated={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
